nxc/protocols/ssh.py

```diff
@@ -111,76 +209,82 @@ def plaintext_login(self, username, password, private_key=None):
                     cred_id = self.db.add_credential(
                         "key",
                         username,
-                        password if password != "" else "",
+                        "",
                         key=key_data,
                     )
             else:
-                self.logger.debug(f"Logging in with password")
-                self.conn.connect(
-                    self.host,
-                    port=self.args.port,
-                    username=username,
-                    password=password,
-                    look_for_keys=False,
-                    allow_agent=False,
-                )
-                cred_id = self.db.add_credential("plaintext", username, password)
+                self.logger.debug(f"Logging {self.host} with username: {self.username}, password: {self.password}")
+                self.conn._transport.auth_password(username, password, fallback=True)
 
+            # Some IOT devices will not raise exception in self.conn._transport.auth_password / self.conn._transport.auth_publickey
```

Comment:
Which IoT devices?

Reply:
I test it with my home network switch (Netgear GS308T)